src/PointwiseFunctions/GeneralRelativity/ComputeGhQuantities.cpp

```diff
@@ -166,6 +167,250 @@ tnsr::ijj<DataType, SpatialDim, Frame> deriv_spatial_metric(
   return deriv_spatial_metric;
 }
 
+template <size_t SpatialDim, typename Frame, typename DataType>
+tnsr::i<DataType, SpatialDim, Frame> spatial_deriv_of_lapse(
+    const Scalar<DataType>& lapse,
+    const tnsr::A<DataType, SpatialDim, Frame>& spacetime_unit_normal,
+    const tnsr::iaa<DataType, SpatialDim, Frame>& phi) noexcept {
+  auto deriv_lapse =
+    make_with_value<tnsr::i<DataType, SpatialDim, Frame>>(get(lapse), 0.0);
+  for (size_t i = 0; i < SpatialDim; ++i) {
+    for (size_t a = 0; a < SpatialDim+1; ++a) {
+      for (size_t b = 0; b < SpatialDim+1; ++b) {
+        deriv_lapse.get(i) += phi.get(i, a, b)
+          * spacetime_unit_normal.get(a) * spacetime_unit_normal.get(b);
+      }
+    }
+    deriv_lapse.get(i) *= (-0.5 * get(lapse));
+  }
+  return deriv_lapse;
+}
+
+template <size_t SpatialDim, typename Frame, typename DataType>
+Scalar<DataType> time_deriv_of_lapse(
+    const Scalar<DataType>& lapse,
+    const tnsr::I<DataType, SpatialDim, Frame>& shift,
+    const tnsr::A<DataType, SpatialDim, Frame>& spacetime_unit_normal,
+    const tnsr::iaa<DataType, SpatialDim, Frame>& phi,
+    const tnsr::aa<DataType, SpatialDim, Frame>& pi) noexcept {
+  auto dt_lapse = make_with_value<Scalar<DataType>>(get(lapse), 0.);
+  for (size_t a = 0; a < SpatialDim+1; ++a) {
+    for (size_t b = 0; b < SpatialDim+1; ++b) {
+      // first term
+      get(dt_lapse) += 0.5 * get(lapse) * get(lapse) * pi.get(a, b)
+            * spacetime_unit_normal.get(a) * spacetime_unit_normal.get(b);
+      // second term
+      for (size_t i = 0; i < SpatialDim; ++i) {
+        get(dt_lapse) -= 0.5 * get(lapse) * shift.get(i)
+            * spacetime_unit_normal.get(a) * spacetime_unit_normal.get(b)
+            * phi.get(i, a, b);
+      }
+    }
+  }
+  return dt_lapse;
+}
+
+template <size_t SpatialDim, typename Frame, typename DataType>
+tnsr::iJ<DataType, SpatialDim, Frame> spatial_deriv_of_shift(
+    const Scalar<DataType>& lapse,
+    const tnsr::AA<DataType, SpatialDim, Frame>& inverse_spacetime_metric,
+    const tnsr::A<DataType, SpatialDim, Frame>& spacetime_unit_normal,
+    const tnsr::iaa<DataType, SpatialDim, Frame>& phi) noexcept {
+  auto deriv_shift =
+    make_with_value<tnsr::iJ<DataType, SpatialDim, Frame>>(get(lapse), 0.);
+  for (size_t i = 0; i < SpatialDim; ++i) {
+    for (size_t j = 0; j < SpatialDim; ++j) {
+      for (size_t a = 0; a < SpatialDim+1; ++a) {
```

Comment:
I think `a` starting from 1 will be sufficient.

Reply:
Functions computing derivs of 3plus1 quantities have been moved to  #1176 , #1177 , #1179 . Please review there.